<launch>

    <group ns="stereo_vision" clear_params="true">
        <arg name="camera_config" value="OV5647_MIPI_2Lane_RAW8_8b_2592x1944_BA.cfg" /> <!-- Config file in camera_config_files folder of your camera. -->
        <arg name="calibration_left" default="left_cam.yaml" />
        <arg name="calibration_right" default="right_cam.yaml" />


        <node name="stereo_vision_node" pkg="stereo_vision_ros" type="stereo_vision_node.py" output="screen">
            <param name="config_file" value="$(find stereo_vision_ros)/camera_config_files/$(arg camera_config)" />
            <param name="calibration_left" value="$(find stereo_vision_ros)/calib/$(arg calibration_left)" />
            <param name="calibration_right" value="$(find stereo_vision_ros)/calib/$(arg calibration_right)" />
        </node>

        <node name="stereo_rectifier_node" pkg="stereo_vision_ros" type="stereo_rectifier_node.py" output="screen" />

        <node name="stereo_depth_node" pkg="stereo_vision_ros" type="stereo_depth_node.py" output="screen" />

        <node name="stereo_touch_node" pkg="stereo_vision_ros" type="stereo_touch_node.py" output="screen">
            <param name="hsvValues" value="$(find stereo_vision_ros)/calib/hsvValues" />
        </node>

        <node type="rviz" name="rviz" pkg="rviz" args="-d $(find stereo_vision_ros)/launch/rviz_pointCloud_config.rviz" />

    </group>

</launch>
